pkg/operator/ceph/file/health.go

```diff
@@ -0,0 +1,113 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package file manages a CephFS filesystem and the required daemons.
+package file
+
+import (
+	"time"
+
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/clusterd"
+	cephclient "github.com/rook/rook/pkg/daemon/ceph/client"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	defaultHealthCheckInterval = 1 * time.Minute
+)
+
+type mirrorChecker struct {
+	context        *clusterd.Context
+	interval       time.Duration
+	client         client.Client
+	clusterInfo    *cephclient.ClusterInfo
+	namespacedName types.NamespacedName
+	fsSpec         *cephv1.FilesystemSpec
+	fsName         string
+}
+
+// newMirrorChecker creates a new HealthChecker
+func newMirrorChecker(context *clusterd.Context, client client.Client, clusterInfo *cephclient.ClusterInfo, namespacedName types.NamespacedName, fsSpec *cephv1.FilesystemSpec, fsName string) *mirrorChecker {
+	c := &mirrorChecker{
+		context:        context,
+		interval:       defaultHealthCheckInterval,
+		clusterInfo:    clusterInfo,
+		namespacedName: namespacedName,
+		client:         client,
+		fsSpec:         fsSpec,
+		fsName:         fsName,
+	}
+
+	// allow overriding the check interval
+	checkInterval := fsSpec.StatusCheck.Mirror.Interval
+	if checkInterval != nil {
+		logger.Infof("filesystem %q mirroring status check interval is %q", namespacedName.Name, checkInterval)
+		c.interval = checkInterval.Duration
+	}
+
+	return c
+}
+
+// checkMirroring periodically checks the health of the cluster
+func (c *mirrorChecker) checkMirroring(stopCh chan struct{}) {
+	// check the mirroring health immediately before starting the loop
+	err := c.checkMirroringHealth()
+	if err != nil {
+		c.updateStatusMirroring(nil, err.Error())
+		logger.Debugf("failed to check filesystem mirroring status %q. %v", c.namespacedName.Name, err)
+	}
+
+	for {
+		select {
+		case <-stopCh:
+			logger.Infof("stopping monitoring filesystem mirroring status %q", c.namespacedName.Name)
+			return
+
+		case <-time.After(c.interval):
+			logger.Debugf("checking filesystem mirroring status %q", c.namespacedName.Name)
+			err := c.checkMirroringHealth()
+			if err != nil {
+				c.updateStatusMirroring(nil, err.Error())
+				logger.Debugf("failed to check filesystem %q mirroring status. %v", c.namespacedName.Name, err)
+			}
+		}
+	}
+}
+
+func (c *mirrorChecker) checkMirroringHealth() error {
+	mirrorStatus, err := cephclient.GetFSMirrorDaemonStatus(c.context, c.clusterInfo, c.fsName)
+	if err != nil {
+		c.updateStatusMirroring(nil, err.Error())
+		return err
+	}
+
+	// TODO: uncomment me once this is fixed: https://github.com/rook/rook/issues/7975
```

Comment:
Are we just waiting for 16.2.5 for this?

Reply:
Not really, the fix went into master and did not get backport as far as I can tell. I need to test a different approach.

Reply:
FYI it's fixed and I'm using it now.